### PR TITLE
feat(react-table, react-components): export DataGridContextProvider

### LINKED ISSUE #28944: [Feature]: Export `DataGrid` context.
### Library

React Components / v9 (@fluentui/react-components)

### Describe the feature that you would like added

Hi,

We need to extend the DataGrid component. The change requires a new render function but the library does not export the `DataGridContextProvider`. Can the `DataGridContextProvider` be exported from the library? For example `Table` exports its `TableContextProvider` [see](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-table/src/index.ts#L93).

Thanks!

Thanks for the great library!

### Have you discussed this feature with our team

_No response_

### Additional context

_No response_

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Priority

Normal

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

DataGridContextProvider is not exported form its package and react-components.

## New Behavior

DataGridContextProvider is now exported form its package and react-components.

## Related Issue(s)

- Fixes #28944

### CODE REVIEW

**Comment by sopranopillow** [comment]
@ling1726, any reason not to do this? Seems reasonable since we export everything else in the context.

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/InfoButton-mount.html">12</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Avatar-mount_base.html">600</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Avatar-mount.html">596</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Button-mount_base.html">306</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Button-mount.html">307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Field-mount_base.html">1082</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Field-mount.html">1107</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProvider-mount_base.html">682</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProvider-mount.html">673</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProviderWithTheme-mount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProviderWithTheme-mount.html">87</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">67</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/InfoButton-mount.html">12</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/MakeStyles-mount_base.html">884</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/MakeStyles-mount.html">847</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Persona-mount_base.html">1698</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/Persona-mount.html">1654</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/SpinButton-mount_base.html">1328</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28955/perf-test-react-components/SpinButton-mount.html">1309</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by khmakoto — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `66.614 kB`<br />`18.512 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `203.567 kB`<br />`57.847 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `37.787 kB`<br />`12.387 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |     `6.48 kB`<br />`2.203 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 |  `152.77 kB`<br />`42.242 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `39.638 kB`<br />`12.129 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `126.224 kB`<br />`33.612 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |   `71.594 kB`<br />`18.91 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `70.213 kB`<br />`18.511 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/0c71b67bb41fdc2de8c3c547e77646a4b2ab36e9'>0c71b67bb41fdc2de8c3c547e77646a4b2ab36e9</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-table/src/index.ts [hunk 1]
```diff
@@ -163,6 +163,7 @@ export type {
   DataGridContextValue,
   DataGridFocusMode,
 } from './DataGrid';
+export { DataGridContextProvider } from './contexts/dataGridContext';
```
Comment (ling1726) [inline packages/react-components/react-table/src/index.ts:166]: you can also export the hook to use the context, as long as we're not exporting the context directly it's fine. And in theory to allow recomposition, we should export both

**Review by ling1726 — APPROVED** [review]

**Review by ling1726 — APPROVED** [review]

**Pull request merged**